compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6407,7 +6407,6 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
 
         BType retType = typeParamAnalyzer.getReturnTypeParams(env, bInvokableType.getReturnType());
         if (restType != symTable.semanticError &&
-                Symbols.isFlagOn(invokableSymbol.flags, Flags.NATIVE) &&
```

Comment:
Alternatively I guess we can check if it is a method declaration, but I guess this isn't quite necessary since we are checking if the return type is "parameterized".

```java
Symbols.isFlagOn(invokableSymbol.flags, Flags.NATIVE) || 
   Symbols.isFlagOn(invokableSymbol.flags, Flags.INTERFACE) 
```

Reply:
Thanks. I think it is better to be expressive in code for anyone to understand. I will check this logic.